### PR TITLE
Disregard incoming remote target on receiving links initiated by the plugin

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by k-wall** [comment]
@lulf unfortunately I needed to use some java package trickery to get the access I needed to disregard the incoming attach source/targets for initiated links.  I don't see a neater way.

In the longer term, I think Artemis plugin connection plugin needs to somehow signal back to Artemis that the link was self-initiated, so that the target (of receiving link) and the source (of sending links) on the attach response can be ignored.  I think an easy way to do this is to add an Qpid Proton Attachment to the Link.  This can be used to allow Artemis to distinguish a self initiated link from one that was initiated by the partner.

https://qpid.apache.org/releases/qpid-proton-j-0.33.4/api/org/apache/qpid/proton/engine/Extendable.html

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=isolated

File: broker-plugin/plugin/src/main/resources/logging.properties [hunk 1]
```diff
@@ -20,9 +20,9 @@
 loggers=org.eclipse.jetty,org.jboss.logging,org.apache.activemq.artemis.core.server,org.apache.activemq.artemis.utils,org.apache.activemq.artemis.journal,org.apache.activemq.artemis.jms.server,org.apache.activemq.artemis.integration.bootstrap,org.apache.activemq.audit.base,org.apache.activemq.audit.message
 
 # Root logger level
-logger.level=INFO
+logger.level=DEBUG
```
Comment (lulf) [inline broker-plugin/plugin/src/main/resources/logging.properties:23]: Should probably switch this back.

Comment (k-wall) [reply broker-plugin/plugin/src/main/resources/logging.properties]: Fixed

**Comment by k-wall** [comment]
@lulf I addressed the review comment and made a few more tidy ups.  I also introduced a env-var that could be used to turn off the change.

**Pull request merged**